terra_notebook_utils/__init__.py

```diff
@@ -26,3 +28,20 @@
     DRS_RESOLVER_URL = MARTHA_URL
 else:
     DRS_RESOLVER_URL = DRSHUB_URL
+
+
+class ExecutionEnvironment(Enum):
+    TERRA_WORKSPACE = 1,  # Executing in a Terra Workspace (on any supported platform)
+    OTHER = 2  # Executing outside of a Terra Workspace (e.g., local system)
+
+
+class ExecutionPlatform(Enum):
+    AZURE = 1,  # Executing in an Azure compute environment
+    GOOGLE = 2,  # Executing in a Google compute environment
+    UNKNOWN = 3  # Execution platform not identified (e.g., local system)
```

Comment:
Nit:  In a debugger, these are kind of magic numbers.  It might be clearer to simply use strings (i.e. "google") instead of numbers.  This also has the benefit that a user can print "ExecutionContext.execution_platform" and see "google" or "azure" instead of "2" or "1".

Reply:
Good idea! Enum values are now (uppercase) string values.